### DISCUSSION TITLE
rbenv asing for 2.7.0 when version 3.0.3 is installed

### DISCUSSION THREAD

**Discussion by cecileRx** [rbenv/ruby-build#1959] [Build failures] [answered]
Hello Everyone! 
Can someone explain why rbenv still ask for version '2.7.0' when I got installed the '3.0.3' version? 
(this keeps me from installing Rails)

here are my terminal logs:

```
brew uninstall --force rbenv ruby-build
exec zsh
➜  ~ brew install rbenv
==> Downloading https://ghcr.io/v2/homebrew/core/autoconf/manifests/2.71
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/autoconf/blobs/sha256:258a94bef
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sh
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/pkg-config/manifests/0.29.2_3
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/pkg-config/blobs/sha256:80f141e
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sh
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/ruby-build/manifests/20220415
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/ruby-build/blobs/sha256:b856737
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sh
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/rbenv/manifests/1.2.0
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/rbenv/blobs/sha256:a2ca52c4fe3b
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sh
######################################################################## 100.0%
==> Installing dependencies for rbenv: autoconf, pkg-config and ruby-build
==> Installing rbenv dependency: autoconf
==> Pouring autoconf--2.71.catalina.bottle.tar.gz
🍺  /usr/local/Cellar/autoconf/2.71: 71 files, 3.2MB
==> Installing rbenv dependency: pkg-config
==> Pouring pkg-config--0.29.2_3.catalina.bottle.tar.gz
🍺  /usr/local/Cellar/pkg-config/0.29.2_3: 11 files, 624.0KB
==> Installing rbenv dependency: ruby-build
==> Pouring ruby-build--20220415.all.bottle.tar.gz
🍺  /usr/local/Cellar/ruby-build/20220415: 563 files, 281.2KB
==> Installing rbenv
==> Pouring rbenv--1.2.0.catalina.bottle.tar.gz
🍺  /usr/local/Cellar/rbenv/1.2.0: 35 files, 106.8KB
==> Running `brew cleanup rbenv`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
➜  ~ rbenv install 3.0.3
Downloading openssl-1.1.1n.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/40dceb51a4f6a5275bde0e6bf20ef4b91bfc32ed57c0552e2e8e15463372b17a
Installing openssl-1.1.1n...
Installed openssl-1.1.1n to /Users/rougnaux/.rbenv/versions/3.0.3

Downloading ruby-3.0.3.tar.gz...
-> https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.3.tar.gz
Installing ruby-3.0.3...
ruby-build: using readline from homebrew
Installed ruby-3.0.3 to /Users/rougnaux/.rbenv/versions/3.0.3

➜  ~ rbenv global 3.0.3
➜  ~ exec zsh
➜  ~ ruby -v
rbenv: version `2.7.0' is not installed (set by /Users/rougnaux/.ruby-version)
```

**Comment by mislav** [comment, marked as answer]
Hi, I spotted the problem by looking at this line:
```
rbenv: version `2.7.0' is not installed (set by /Users/rougnaux/.ruby-version)
```
It looks like you have a `/Users/rougnaux/.ruby-version` file, which sets the "local" Ruby version to `2.7.0`. This was likely created by running `rbenv local 2.7.0` while in your home directory. This is probably not what you want, since any sub-directory of `/Users/rougnaux` (basically, everything under your user) will inherit that version, but you want your "global" version to be `3.0.3`.

You should `rm /Users/rougnaux/.ruby-version` to fix this.

"Local" ruby version are meant primarily for project directories. For example, you can run `rbenv local 2.7.0` to set it as a local Ruby version while in the directory `/Users/rougnaux/projects/myapp`. Then, everything under that directory will use Ruby 2.7.0. However, it's probably not a good idea to run `rbenv local ...` in your home directory.

**Comment by cecileRx** [comment]
Hello Mislav,
Thanks for your detailed and clear answer. I am wondering how this file could have been created 😉
Anyway, I found a (bad) solution in opening this file and changing the line by 3.0.3.
But as you say it's an absurd idea to have a local version in my home directory, so I will remove it!
Thanks again,
C
________________________________
De : Mislav Marohnić ***@***.***>
Envoyé : lundi 18 avril 2022 10:00
À : rbenv/ruby-build ***@***.***>
Cc : Cécile Rougnaux ***@***.***>; Author ***@***.***>
Objet : Re: [rbenv/ruby-build] rbenv asing for 2.7.0 when version 3.0.3 is installed (Discussion #1959)


Hi, I spotted the problem by looking at this line:

rbenv: version `2.7.0' is not installed (set by /Users/rougnaux/.ruby-version)


It looks like you have a /Users/rougnaux/.ruby-version file, which sets the "local" Ruby version to 2.7.0. This was likely created by running rbenv local 2.7.0. This is probably not what you want, since any sub-directory of /Users/rougnaux (basically, everything under your user) will inherit that version, but you want your "global" version to be 3.0.3.

You should rm /Users/rougnaux/.ruby-version to fix this.

"Local" ruby version are meant primarily for project directories. For example, you can run rbenv local 2.7.0 to set it as a local Ruby version while in the directory /Users/rougnaux/projects/myapp. Then, everything under that directory will use Ruby 2.7.0. However, it's probably not a good idea to run rbenv local ... in your home directory.
